lib/spack/env/cc

```diff
@@ -108,6 +108,12 @@ case "$command" in
         ;;
     ld)
         mode=ld
+        for arg in "$@"; do
+            if [ "$arg" = -S ]; then
```

Comment:
Somehow when using the --with-build-option=bootstrap-debug "ld -S -r" is invoked and spack intercepts this. Here is an excerpt from the spack debug log:

in:	ld	-S -r -no_uuid /private/var/folders/hm/7pxrf4vx4_g4f_nkfrcwqq8m0000gp/T/gartung/spack-stage/spack-stage-PZv9zT/spack-build/stage2-x86_64-apple-darwin14.5.0/libstdc++-v3/src/compatibility.o -o /private/var/folders/hm/7pxrf4vx4_g4f_nkfrcwqq8m0000gp/T/gartung/spack-stage/spack-stage-PZv9zT/spack-build/stage2-x86_64-apple-darwin14.5.0/libstdc++-v3/src/compatibility.o.stripped

out:	as	ld -L/Users/gartung/spack/opt/spack/darwin-x86_64/clang-7.0.2-apple/mpfr-3.1.4-bvgor6plszncasuc2wtpo6uuuoq23wxt/lib -rpath /Users/gartung/spack/opt/spack/darwin-x86_64/clang-7.0.2-apple/mpfr-3.1.4-bvgor6plszncasuc2wtpo6uuuoq23wxt/lib -L/Users/gartung/spack/opt/spack/darwin-x86_64/clang-7.0.2-apple/mpc-1.0.3-jhigwld6okg5lvlvzugnm43nchbgzq2m/lib -rpath /Users/gartung/spack/opt/spack/darwin-x86_64/clang-7.0.2-apple/mpc-1.0.3-jhigwld6okg5lvlvzugnm43nchbgzq2m/lib -L/Users/gartung/spack/opt/spack/darwin-x86_64/clang-7.0.2-apple/libiconv-1.14-6334hosj4o4mzeij5wgzebuzfywfp74a/lib -rpath /Users/gartung/spack/opt/spack/darwin-x86_64/clang-7.0.2-apple/libiconv-1.14-6334hosj4o4mzeij5wgzebuzfywfp74a/lib -L/Users/gartung/spack/opt/spack/darwin-x86_64/clang-7.0.2-apple/gmp-6.1.0-kmnkwalnopisi4yxpk6b5fspre3cqlps/lib -rpath /Users/gartung/spack/opt/spack/darwin-x86_64/clang-7.0.2-apple/gmp-6.1.0-kmnkwalnopisi4yxpk6b5fspre3cqlps/lib -S -r -no_uuid /private/var/folders/hm/7pxrf4vx4_g4f_nkfrcwqq8m0000gp/T/gartung/spack-stage/spack-stage-PZv9zT/spack-build/stage2-x86_64-apple-darwin14.5.0/libstdc++-v3/src/compatibility.o -o /private/var/folders/hm/7pxrf4vx4_g4f_nkfrcwqq8m0000gp/T/gartung/spack-stage/spack-stage-PZv9zT/spack-build/stage2-x86_64-apple-darwin14.5.0/libstdc++-v3/src/compatibility.o.stripped